homeworks/yuriy.chaban_YuriyChaban/homework_2/hw1.js

```diff
@@ -0,0 +1,27 @@
+const readline = require('readline');
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+});
+
+rl.question('Please, indicate the height of the pyramid (number of lines) as a number greater than 0:\n', (answer) => {
+    const totalNumberOfRows = parseInt(Math.ceil(answer), 10);
+    const isAnswerIsNumber = typeof totalNumberOfRows === 'number' && !Number.isNaN(totalNumberOfRows);
+
+    if (isAnswerIsNumber && totalNumberOfRows > 0) {
```

Comment:
IMHO, you can reverse validation of input and exit from a function if something wrong. Therefore your main business code will be but splitted from the validation part.

Reply:
(answer) => {
    if (!valid1) {
        console.error('Some error 1');
        return;
    }

    if (!valid2) {
        console.error('Some error 2');
        return;
    }

    businessLogic();
}

Reply:
I can divide the conditions of the calculation! 
Need I do this?